### PR TITLE
Use new metadata syntax

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
Updates the metadata to use the new syntax of `language:directive(arguments)`.
I temporarily added a function for converting between the old and new syntax. It's only temporary, and will be removed in my next PR to finish my metadata changes.

Also some general metadata cleanup:
- In some files we would explicitly declare `[[cpp:header-ext:h]]`, but using a `.h` extension is already the default, so I removed these directives.
- Removed language specific directives where they didn't make sense.
- Removed leftovers from metadata we've removed (either bogus usage in a Slice file, or vestigial code in the compilers)

### CODE REVIEW

**Comment by pepone** [comment]
> In some files we would explicitly declare [[cpp:header-ext:h]], but using a .h extension is already the default, so I removed these directives.

I think this is still required, a user might be running the compiler with --header-ext hh, but the generated includes for Ice still use Current.h

**Comment by InsertCreativityHere** [comment]
Yes, I suppose that's true. I'll add them back in!

**Comment by pepone** [comment]
> Yes, I suppose that's true. I'll add them back in!
Or maybe is time to remove `--header-ext`?

**Comment by bernardnormier** [comment]
> > Or maybe is time to remove `--header-ext`?

Yes, we should remove `--header-ext` and various other deprecated command-line arguments. This way, metadata with a default value that can no longer be overridden becomes redundant.

File: csharp/test/Ice/compress/Test.ice [hunk 1]
```diff
@@ -18,9 +18,9 @@ exception MyException
 
 interface TestIntf
 {
-    [compress:params] void opCompressParams(int size, ByteSeq p1);
-    [compress:return] ByteSeq opCompressReturn(int size);
-    [compress:params,return] ByteSeq opCompressParamsAndReturn(ByteSeq p1);
+    [compress(params)] void opCompressParams(int size, ByteSeq p1);
+    [compress(return)] ByteSeq opCompressReturn(int size);
+    [compress(params,return)] ByteSeq opCompressParamsAndReturn(ByteSeq p1);
```
Comment (bernardnormier) [inline csharp/test/Ice/compress/Test.ice:23]: I think it should be formatted like a function call:
```
[compress(params, return)]
```

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: csharp/test/Ice/compress/Test.ice (hunk 1 above)
Comment (InsertCreativityHere) [reply csharp/test/Ice/compress/Test.ice]: Fixed! I also checked everywhere else we used commas in metadata, and this was the only place we did (apart from stuff like `cpp:type(map<int, int>)`)

**Pull request merged**